### PR TITLE
u256 more operations

### PR DESCRIPTION
**Pull request by xunilrj**
## Description

This PR is part of https://github.com/FuelLabs/sway/pull/4794 and implements more operators for u256:
- [x] sub
- [x] mul
- [x] div
- [x] comparisions
- [x] mod

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: sway-ir/src/optimize/misc_demotion.rs [hunk 1]
```diff
@@ -445,39 +451,245 @@ fn wide_binary_op_demotion(context: &mut Context, function: Function) -> Result<
...
+
+        let cmp_op_metadata = cmp_instr_val.get_metadata(context);
+
+        let arg1_ty = arg1.get_type(context).unwrap();
+        let arg1_metadata = arg1.get_metadata(context);
+        let arg2_ty = arg2.get_type(context).unwrap();
+        let arg2_metadata = arg2.get_metadata(context);
+
+        // If arg1 is not a pointer, store it to a local
+        let lhs_store = if !arg1_ty.is_ptr(context) {
```
Comment (IGI-111) [inline sway-ir/src/optimize/misc_demotion.rs:608]: You can use [`then`](https://doc.rust-lang.org/std/primitive.bool.html#method.then) here

File: sway-ir/src/optimize/misc_demotion.rs [hunk 2]
```diff
@@ -445,39 +451,245 @@ fn wide_binary_op_demotion(context: &mut Context, function: Function) -> Result<
...
+        };
+
+        let (arg1_needs_insert, get_arg1) = if let Some((lhs_local, _)) = &lhs_store {
+            (false, *lhs_local)
+        } else {
+            (true, arg1)
+        };
+
+        // If arg2 is not a pointer, store it to a local
+        let rhs_store = if !arg2_ty.is_ptr(context) {
```
Comment (IGI-111) [inline sway-ir/src/optimize/misc_demotion.rs:638]: Same here

**Comment by IGI-111** [comment]
Just some nitpicks, otherwise LGTM

**Comment by vaivaswatha** [comment]
We need IR parsing tests for the new IR opcodes (for this and the recently introduced ones too). I'm guessing that they can go into `sway-ir/tests/serialize`.

File: sway-ir/src/optimize/misc_demotion.rs (hunk 1 above)
Comment (xunilrj) [reply sway-ir/src/optimize/misc_demotion.rs]: Done.

File: sway-ir/src/optimize/misc_demotion.rs (hunk 2 above)
Comment (xunilrj) [reply sway-ir/src/optimize/misc_demotion.rs]: Done.

**Comment by xunilrj** [comment]
> We need IR parsing tests for the new IR opcodes (for this and the recently introduced ones too). I'm guessing that they can go into `sway-ir/tests/serialize`.

Done.

**Review by vaivaswatha — APPROVED** [review]

**Review by IGI-111 — APPROVED** [review]

**Pull request merged**